### PR TITLE
adding .rst to docs/cdk

### PR DESCRIPTION
**Pull request by GFMoraga**
# Change Summary

## Overview
UPDATES  with team suggestions/feedback!

Adding documentation for aws and cdk setup. Wanted to get this going but WILL be updated and worked on more!
 

## New Files
-made new directory in doc called cdk
> added aws-setup.rst 
> added cdk-deployment.rst  
>index.rst

### CODE REVIEW

File: docs/source/cdk/aws-setup.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+****
+aws setup
+****
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]:
```suggestion
AWS setup
=========
```
You need to underline at least as long as the text itself.

File: docs/source/cdk/aws-setup.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+**Download/requirements:**
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]:
```suggestion
Download/requirements
~~~~~~~~~~~~~~~~~~~~~~~
```
Make this a heading too, then it will show up in table of contents.

File: docs/source/cdk/aws-setup.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Ensure you have installed nodejs (newer than version 14), AWS CLI, and Docker
+
+- `nodejs <https://nodejs.org/en/download/>`_
+
+- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
+
+- `Docker <https://docs.docker.com/get-docker/>`_
+
+**New user**
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]: Heading too

File: docs/source/cdk/aws-setup.rst [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+****
+**Download/requirements:**
+
+Ensure you have installed nodejs (newer than version 14), AWS CLI, and Docker
+
+- `nodejs <https://nodejs.org/en/download/>`_
+
+- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
+
+- `Docker <https://docs.docker.com/get-docker/>`_
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]: I don't think there should be a space between the entries.

File: docs/source/cdk/aws-setup.rst [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- click your user button
+- select *Security Credentials* tab
+- click *Create Access Key*
+- make note of *Access Key ID* and *Secret Access Key*
+- you can download the .csv file if you want
+- click *Close*
+
+
+
+
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]: Only need one blank line at the end.

File: docs/source/cdk/cdk-deployment.rst [hunk 6]
```diff
@@ -0,0 +1,11 @@
+************
+cdk deployment
+************
+
+In command line, run the following command to set up the aws environment:
+
+.. code-block::
+    aws configure
```
Comment (greglucas) [inline docs/source/cdk/cdk-deployment.rst]: This is a one-time step, so I think it would make more sense over in the aws-setup portion. It is still very aws-centric.

Here I think you should put all of the content from the README that already exists and indicate how to synthesize and deploy with the cdk. Also adding in numbered steps and commands.

**Review by greglucas — COMMENTED** [review]
I don't think some of the sections will render quite right yet. Were you able to build the docs and see this locally? (let us know if the doc build/open instructions weren't clear and we can definitely update)

Here is a helpful formatting guide for some of my comments: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html

Also, these new pages need to be added to a Table of Contents (TOC). Which there is currently only one up in the `index.rst` file a few levels above these. This page has some information for the project and how it should be organized in a hierarchy: https://imap-processing.readthedocs.io/en/latest/development/doc-overview.html#use-document-hierarchy-to-organize-information

File: docs/source/cdk/cdk-deployment.rst (hunk 6 above)
Comment (GFMoraga) [reply docs/source/cdk/cdk-deployment.rst]: Okay. Sounds good!

**Comment by bourque** [comment]
I just wanted to note here that this PR does not quite adhere to the nominal forking workflow. These changes are taking place on @GFMoraga's `development` branch, but ideally they should be on a feature branch (named something like `add-cdk-docs` or something like that). This is described in step 6 [here](https://imap-processing.readthedocs.io/en/latest/development/style-guide.html#git-and-github-workflow).

I think this is fine for this time around, and sorry to be a stickler about this. I think it is just better practice to be making feature branches, and I find that it can help avoid some painful merging/rebasing/pulling down the line.

**Comment by greglucas** [comment]
@bourque I completely agree and that should be added to our CI checks too then :) Would you mind opening a PR to add that to the pre-commit checks? https://github.com/pre-commit/pre-commit-hooks#no-commit-to-branch

**Comment by bourque** [comment]
> @bourque I completely agree and that should be added to our CI checks too then :) Would you mind opening a PR to add that to the pre-commit checks? https://github.com/pre-commit/pre-commit-hooks#no-commit-to-branch

Ah cool! Yes, I can do that.

File: docs/source/cdk/cdk-deployment.rst (hunk 6 above)
Comment (maxinelasp) [inline docs/source/cdk/cdk-deployment.rst]: If you're using a `code block` directive, you need to include a language: 
``` 
.. code-block:: bash

    aws configure
```
Notice the line break between the code block part and the code. The indentation is correct.

However, if you just want a code block without syntax highlighting, you can just use the double colon. In this case: 
```
In the command line, run the following command:: 

    aws configure
```

So either way, you need a line break in between the double colon and the actual code block. With the `code-block` directive, you also need to provide it with a language (here, bash). 

You can see these errors including the ones Greg already pointed out above by following the README.md guide under docs/ to install and run sphinx locally. If you need help with poetry, let me know.

File: docs/source/cdk/aws-setup.rst [hunk 7]
```diff
@@ -0,0 +1,33 @@
+****
+aws setup
```
Comment (maxinelasp) [inline docs/source/cdk/aws-setup.rst]: Capitalize titles (AWS Setup)

File: docs/source/cdk/aws-setup.rst [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+- `nodejs <https://nodejs.org/en/download/>`_
+
+- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
+
+- `Docker <https://docs.docker.com/get-docker/>`_
+
+**New user**
+
+- log-in to AWS console
+`AWS console <https://aws.amazon.com/console/>`_
```
Comment (maxinelasp) [inline docs/source/cdk/aws-setup.rst]:
```suggestion
- log-in to `AWS console <https://aws.amazon.com/console/>`_
```

File: docs/source/cdk/aws-setup.rst [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+- enter user name and password
+
+- select IAM from *Services* menu or search IAM
+- select *Users* from left menu
+- click your user button
+- select *Security Credentials* tab
+- click *Create Access Key*
+- make note of *Access Key ID* and *Secret Access Key*
+- you can download the .csv file if you want
+- click *Close*
```
Comment (maxinelasp) [inline docs/source/cdk/aws-setup.rst]: Nitpick, but can you make this a numbered list?

You can have the numbers automatically generated by using `#.` instead of `-`

**Review by maxinelasp — CHANGES_REQUESTED** [review]
Overall, this is a good first pass. There are a few errors when building (which Greg pointed out), and it needs to be added to an index file. 

To do this, create a new `index.rst` file under `docs/source/cdk`. This will act as the landing page for all the CDK docs. Then, add a `toctree` block to that. You can look to [this](https://raw.githubusercontent.com/IMAP-Science-Operations-Center/imap_processing/dev/docs/source/development/index.rst) index.rst file to see how the whole thing should look. 

Basically, it requires a block that should look like this at the end:

```
.. toctree::
    :maxdepth: 1

    aws-setup
    cdk-deployment
```

Then, update the `index.rst` file under `docs/source` to include your new `index.rst` file for cdk. 

Before:
```
.. toctree::
   :maxdepth: 2
   :caption: Contents:
```

After:
```
.. toctree::
   :maxdepth: 2
   :caption: Contents:

   docs/index
```

This will allow all the documents to link to each other, like on the `imap_processing` docs.

File: docs/source/cdk/aws-setup.rst [hunk 10]
```diff
@@ -1,33 +1,43 @@
...
+#. For region, set it to the AWS region you'd like to set up your SDS. For IMAP, we're using "us-west-2"::
+
+    [imap]
+    region=us-west-2
+    aws_access_key_id=<Access Key>
+    aws_secret_access_key=<Secret Key>
+
+#. Then, you can set the profile used by cdk by setting the AWS_PROFILE environment variable to the profile name (in this case, imap)::
+
+    export AWS_PROFILE=imap
```
Comment (maxinelasp) [inline docs/source/cdk/aws-setup.rst:47]: Nice! This whole file looks loads better

File: docs/source/cdk/cdk-deployment.rst [hunk 11]
```diff
@@ -1,11 +1,30 @@
...
-In command line, run the following command to set up the aws environment:
+#. AWS_PROFILE ()
+#. Your initials when deploying to an AWS account with multiple users ()
 
-.. code-block::
-    aws configure
+To deploy, first set the appropriate environment variables::
 
+        export AWS_PROFILE=<profile>
+You'll then need to synthesize the CDK code with the command::
```
Comment (maxinelasp) [inline docs/source/cdk/cdk-deployment.rst:16]: This needs a blank line in between the block and the next paragraph
```suggestion

You'll then need to synthesize the CDK code with the command::
```

File: docs/source/cdk/cdk-deployment.rst [hunk 12]
```diff
@@ -1,11 +1,30 @@
...
 
+        cdk synth --app "python app_template_dev.py"
+Then you can deploy the architecture with the following command::
```
Comment (maxinelasp) [inline docs/source/cdk/cdk-deployment.rst:20]:
```suggestion

Then you can deploy the architecture with the following command::
```

File: docs/source/index.rst [hunk 13]
```diff
@@ -10,3 +10,5 @@ Welcome to sds-data-manager's documentation!
    :maxdepth: 2
    :caption: Contents:
 
+   docs/index
```
Comment (maxinelasp) [inline docs/source/index.rst]:
```suggestion
   cdk/index
```

Sorry if this was a typo from me!

File: docs/source/cdk/index.rst [hunk 14]
```diff
@@ -0,0 +1,5 @@
+.. toctree::
```
Comment (maxinelasp) [inline docs/source/cdk/index.rst:3]: I'd suggest adding a title here just so it's clear what this documentation is. Something like 

```
CDK Deployment 
=============
```

**Review by maxinelasp — APPROVED** [review]
Looks a lot better! Just added a few suggestions to fix some errors. Once those are in, this looks good to me!

**Comment by maxinelasp** [comment]
Oh, and if you moved information out of the readme, you should delete that duplicated info and add a link instead!

File: docs/source/cdk/aws-setup.rst [hunk 15]
```diff
@@ -0,0 +1,43 @@
+AWS Setup
+=========
+Download/requirements
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]:
```suggestion
=========

Download/requirements
~~~~~~~~~~~~~~~~~~~~~
```
Extra linebreak between the sections, and it is nice to make the underlines exactly the same length even though the spec does allow longer.

I would have said an extra linebreak after even the minor section underlines, but it doesn't look like @maxinelasp did that in imap-processing. I would follow her lead for consistency, which I think did include the linebreak between the primary page header.

File: docs/source/cdk/aws-setup.rst [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+#. You can download the .csv file if you want
+#. Click *Close*
+
+Existing User
+~~~~~~~~~~~~~
+#. In command line, run the following command to set up the aws environment::
+
+    aws configure
+
+#. Enter the *Access Key ID* and *Secret Access Key* from the previous step
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]:
```suggestion
#. Enter your *Access Key ID* and *Secret Access Key* (If you don't have these, follow the :ref:`aws-new-user` section)
```
Adding a restructured text link to the previous section would be nice too. https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role

```python
# just above the previous section
.. _aws-new-user:
```

File: docs/source/cdk/cdk-deployment.rst [hunk 17]
```diff
@@ -0,0 +1,30 @@
+CDK deployment
+==============
+Deploy
+~~~~~~
+You will need to make a **copy** of app_template_dev.py file with a different name (app__dev.py) and keep a copy of
```
Comment (greglucas) [inline docs/source/cdk/cdk-deployment.rst]:
```suggestion
~~~~~~
You will need to make a **copy** of :file:`app_template_dev.py` file with a different name :file:`app_<initials>_dev.py` and keep a copy of
```
Thanks for copying this over into the docs! See here for an update of what this should be to avoid the links: https://github.com/IMAP-Science-Operations-Center/sds-data-manager/pull/111
In rst I think you want the file directive to format it indicating it is an external file location.
https://devguide.python.org/documentation/markup/

File: docs/source/cdk/cdk-deployment.rst [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+it locally so that it will not be committed. In your own copy there are two important configuration
+items which you can alter:
+
+#. AWS_PROFILE ()
+#. Your initials when deploying to an AWS account with multiple users ()
```
Comment (greglucas) [inline docs/source/cdk/cdk-deployment.rst]: What does the trailing `()` do? Are you meaning to put ``<profile>``  and ``<initials>`` in as reference here instead.

File: docs/source/cdk/cdk-deployment.rst [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+
+To deploy, first set the appropriate environment variables::
+
+        export AWS_PROFILE=<profile>
+You'll then need to synthesize the CDK code with the command::
+
+        cdk synth --app "python app_template_dev.py"
```
Comment (greglucas) [inline docs/source/cdk/cdk-deployment.rst]:
```suggestion
        cdk synth --app "python app_<initials>_dev.py"
```
I'm always torn on whether to put something runnable in, or not. I don't think we want people accidentally modifying `app_template_dev.py` here though, so would slightly lean towards putting the placeholder in instead?

File: docs/source/cdk/aws-setup.rst (hunk 16 above)
Comment (GFMoraga) [reply docs/source/cdk/aws-setup.rst]: I cant seem to get this. It might be a "my.pars" or something to do with the conf.py file

File: docs/source/cdk/cdk-deployment.rst (hunk 19 above)
Comment (GFMoraga) [reply docs/source/cdk/cdk-deployment.rst]: Agree

**Comment by GFMoraga** [comment]
'pre-commit.ci.autofix'

**Comment by GFMoraga** [comment]
'pre-commit.ci autofix'

File: docs/source/cdk/aws-setup.rst [hunk 20]
```diff
@@ -0,0 +1,45 @@
+AWS Setup
+=========
+Download/requirements
+~~~~~~~~~~~~~~~~~~~~~
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]:
```suggestion
AWS Setup
=========

Download/requirements
~~~~~~~~~~~~~~~~~~~~~
```
Missed the newline here, but added it below. Here is an example of how Numpy's layout is with plenty of newlines between different entities and levels.
https://github.com/numpy/numpy/blob/48a8277855849be094a5979c48d9f5f1778ee4de/doc/source/index.rst?plain=1#L1-L7

File: docs/source/cdk/aws-setup.rst [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+
+Ensure you have installed nodejs (newer than version 14), AWS CLI, and Docker
+
+- `nodejs <https://nodejs.org/en/download/>`_
+- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
+- `Docker <https://docs.docker.com/get-docker/>`_
+
+.. _aws-new-user:
+New User
```
Comment (greglucas) [inline docs/source/cdk/aws-setup.rst]:
```suggestion
.. _aws-new-user:

New User
```
I'd add a newline here as well just to immediately call out this reference link.

File: README.md [hunk 22]
```diff
@@ -1,3 +1,8 @@
+## Requirements
+- AWS CLI [download link](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+- nodejs [download link](https://nodejs.org/en/download/)
+- Docker [download link](https://docs.docker.com/get-docker/)
+
```
Comment (greglucas) [inline README.md:11]: This should go below the main heading and description of the project. (Right after SDS-data-manager and before AWS Setup) Users want to know what the project is before the requirements we need for the project.

File: README.md [hunk 23]
```diff
@@ -48,38 +53,15 @@ If running in codespaces, this should already be done.
...
-aws_access_key_id=<Access Key>
-aws_secret_access_key=<Secret Key>
-```
-
-Then, you can set the profile used by cdk by setting the `AWS_PROFILE` environment variable to the profile name (in this case, imap):
-
-```
-export AWS_PROFILE=imap
-```
+[AWS Setup page](docs/source/cdk/aws-setup.rst)
```
Comment (greglucas) [inline README.md]:
```suggestion
[AWS Setup page](https://sds-data-manager.readthedocs.io/en/latest/cdk/aws-setup.html)
```
I think @maxinelasp was wanting to link out to our built HTML docs instead of the rst which isn't rendered all that nicely on GitHub.

File: README.md [hunk 24]
```diff
@@ -48,38 +53,15 @@ If running in codespaces, this should already be done.
...
 You may also need to set the `CDK_DEFAULT_ACCOUNT` environment variable. 
 
 **NOTE**-- For new AWS users, you'll need to make certain the AWS Cloud Development Kit is installed: 
 
-Ensure you have installed nodejs newer than version 14.
-<https://nodejs.org/en/download/>
 
 ```bash
-nvm use
+nvm use 14
```
Comment (greglucas) [inline README.md]:
```suggestion
nvm use 16
```
14 has dropped off the LTS support list now, so if we are going to recommend something for copy/paste code we should put at least 16 (I think you used 18 fine in the demos?).
https://nodejs.dev/en/about/releases/

**Review by greglucas — COMMENTED** [review]
Looks good, @GFMoraga! Unfortunately doc PRs often get a lot of minor nitpicky comments, so I apologize for those. But you are the first one setting this up, so it is nice to set a good precedent at the beginning. You're doing great addressing all the comments!

It looks like you have a merge conflict now, which I don't think will be too bad to handle, but I think this would be a great opportunity to tag up with @bourque and walk through how he typically handles those and what the steps are for doing conflict resolution during PRs.

**Comment by bourque** [comment]
@GFMoraga We can resolve these conflicts during our tag up today!

File: README.md (hunk 24 above)
Comment (GFMoraga) [reply README.md]: I did use 18. But I can switch it to 16. I think when a new user downloads it auto does the newest version.

Comment (greglucas) [reply README.md]: Yeah, you downloaded "18", but `nvm` == node version manager meaning you can swap around to whatever you want with this command. So you can then select a different one from what you downloaded if you have others on your system. Maybe the best is to put `<version>` here instead of specifying a version? I don't have a strong preference here, other than not 14 so we don't get out of date immediately.

File: README.md (hunk 23 above)
Comment (maxinelasp) [reply README.md]: Yes, links should always go to the rendered version in readthedocs. Inside the rst files you can use either [ref](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#the-ref-role) or [doc](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#the-doc-role) roles.
